### PR TITLE
Implement wipe code feature.

### PR DESCRIPTION
**Pull request by andrewkozlik**

### CODE REVIEW

File: storage/storage.c [hunk 1]
```diff
@@ -1175,6 +1331,36 @@ secbool storage_change_pin(uint32_t oldpin, uint32_t newpin,
...
+  secbool ret = secfalse;
+  if (sectrue == unlock(pin, ext_salt)) {
+    ret = set_wipe_code(wipe_code);
+  }
+  memzero(&pin, sizeof(pin));
+  memzero(&wipe_code, sizeof(wipe_code));
+  return ret;
+}
+
 void storage_wipe(void) {
```
Comment (onvej-sl) [inline storage/storage.c]: You didn't implement this feature:
> When the wipe is triggered, the **EDEK value in the storage will be immediately overwritten with zero bytes**, then norcow_wipe() will be called.

File: storage/storage.c [hunk 2]
```diff
@@ -159,19 +186,22 @@ static secbool secequal(const void *ptr1, const void *ptr2, size_t n) {
   return diff ? secfalse : sectrue;
 }
 
-static secbool secequal32(const uint32_t *ptr1, const uint32_t *ptr2,
-                          size_t n) {
+static secbool secequal32(const void *ptr1, const void *ptr2, size_t n) {
+  assert(n % sizeof(uint32_t) == 0);
```
[the remark this answers is not in the document]
Comment (andrewkozlik) [reply storage/storage.c]: Done.

File: storage/storage.c (hunk 1 above)
Comment (andrewkozlik) [reply storage/storage.c]: This is marked as TODO somewhere. Requires further research, a separate issue created.

File: storage/storage.c [hunk 3]
```diff
@@ -765,39 +926,35 @@ static secbool decrypt_dek(const uint8_t *kek, const uint8_t *keiv) {
...
+  if (sectrue != is_not_wipe_code(pin)) {
+    storage_wipe();
+    error_shutdown("You have entered the", "wipe code. All private",
+                   "data has been erased.", NULL);
+  }
+
   // Get the pin failure counter
   uint32_t ctr = 0;
   if (sectrue != pin_get_fails(&ctr)) {
-    memzero(&pin, sizeof(pin));
```
[the remark this answers is not in the document]
Comment (andrewkozlik) [reply storage/storage.c]: Serves no purpose, but I don't mind keeping it.

File: storage/storage.c [hunk 4]
```diff
@@ -837,7 +994,6 @@ static secbool unlock(uint32_t pin, const uint8_t *ext_salt) {
   if (sectrue != initialized ||
       sectrue != norcow_get(EDEK_PVC_KEY, &rand_salt, &len) ||
       len != RANDOM_SALT_SIZE + KEYS_SIZE + PVC_SIZE) {
-    memzero(&pin, sizeof(pin));
```
[the remark this answers is not in the document]
Comment (andrewkozlik) [reply storage/storage.c]: As above.

**Pull request merged**